node/web/src/lib.rs

```diff
@@ -74,11 +74,57 @@ fn parse_bp_key(key: JsValue) -> Option<AccountSecretKey> {
     )
 }
 
+/// Starts a Mina node in a WASM environment and returns an RPC interface.
+///
+/// This is the main entry point for running a Mina node from JavaScript/WASM.
+/// It spawns the node in a separate thread, sets up all necessary components,
+/// and returns an RPC sender that can be used to communicate with the running
+/// node.
+///
+/// # Arguments
+///
+/// * `block_producer` - Block producer configuration as a JavaScript value.
+///   Can be one of:
+///   - `null`/`undefined`: No block production
+///   - `string`: Plain text secret key
+///   - `[encrypted_key, password]`: Array with encrypted key and password
+/// * `seed_nodes_urls` - Optional list of URLs to fetch peer lists from.
+///   Each URL should return newline-separated peer addresses. This is similar
+///   to the `--peer-list-url` flag in the native node, but allows multiple URLs
+///   to be supplied.
+/// * `seed_nodes_addresses` - Optional list of peer addresses to connect to
+///   directly, in [WebRTC Multiaddr-ish format](https://o1-labs.github.io/mina-rust/docs/developers/webrtc#address-format-differences)
+///   This is directly comparable to the `--peers` flag in the native node.
+/// * `genesis_config_url` - Optional URL to fetch genesis configuration from.
+///   Genesis config must be in bin_prot format. If not provided, uses the default
+///   devnet configuration.
+///
+/// # Returns
+///
+/// An `RpcSender` that can be used to send RPC commands to the running node.
+///
+/// # Panics
```

Comment:
Follow-up: it would be nice to have (automated generated) tests when we make this kind of statements in the documentation.

Reply:
Opened follow-up https://github.com/o1-labs/mina-rust/issues/1862